JavaScript/JavaScriptSDK/SendBuffer.ts

```diff
@@ -183,6 +187,16 @@ module Microsoft.ApplicationInsights {
             var sentElements = this.getBuffer(SessionStorageSendBuffer.SENT_BUFFER_KEY);
             sentElements = sentElements.concat(payload);
 
+            if (sentElements.length > SessionStorageSendBuffer.MAX_BUFFER_SIZE) {
```

Comment:
At the point in time when markAsSent is called isn't it too late to throw away data?